storage/block.go

```diff
@@ -124,27 +122,26 @@ func (b *dbBlock) close() {
 		return
 	}
 
-	// Otherwise, we need to close the stream.
-	if stream := b.stream; stream != nil {
-		b.ctx.RegisterCloser(func() {
-			if bytesPool := b.opts.GetBytesPool(); bytesPool != nil {
-				segment := stream.Segment()
-				stream.Reset(m3db.Segment{})
-				bytesPool.Put(segment.Head)
-				bytesPool.Put(segment.Tail)
-			}
-			stream.Close()
-		})
-	}
+	// Otherwise, we need to return bytes to the bytes pool.
+	segment := b.segment
+	bytesPool := b.opts.GetBytesPool()
+	b.ctx.RegisterCloser(func() {
+		// NB(xichen): we do not return segment tail here because the tails
+		// are precomputed in the encoding options and therefore potentially
+		// reused across different encoders.
```

Comment:
yea good suggestion, will do.

Does tchannel service provide a mechanism for us to know when the bytes have been written to the wire so we can safely return bytes to the pool? Otherwise we can't expire blocks because we don't know when we can't close them. We can make copies of the byte slices but that defeats the purpose of pooling the bytes.